cpp/oneapi/dal/backend/primitives/selection/select_by_rows_quick.hpp

```diff
@@ -0,0 +1,217 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include <limits>
+
+#include <daal/src/externals/service_rng.h>
+#include <daal/include/algorithms/engines/mcg59/mcg59.h>
+#include <daal/src/algorithms/engines/engine_batch_impl.h>
+
+#include "oneapi/dal/backend/primitives/selection/row_partitioning.hpp"
+#include "oneapi/dal/backend/primitives/selection/select_by_rows_base.hpp"
+#include "oneapi/dal/backend/primitives/rnd/rnd_seq.hpp"
+#include "oneapi/dal/backend/primitives/ndarray.hpp"
+
+namespace oneapi::dal::backend::primitives {
+
+#ifdef ONEDAL_DATA_PARALLEL
+
+constexpr std::uint32_t preffered_sg_size = 16;
```

Comment:
It's in hpp, so will be visible for any user of this primitive, suggest to move it to function/class scope or cpp.